packages/react-events/src/Focus.js

```diff
@@ -145,6 +191,57 @@ const FocusResponder = {
       }
     }
   },
+  onRootEvent(
+    event: ReactResponderEvent,
+    context: ReactResponderContext,
+    props: FocusProps,
+    state: FocusState,
+  ): void {
+    const {type, target} = event;
+
+    switch (type) {
+      case 'mousemove':
+      case 'mousedown':
+      case 'mouseup':
+      case 'pointermove':
+      case 'pointerdown':
+      case 'pointerup':
+      case 'touchmove':
+      case 'touchstart':
+      case 'touchend': {
+        if (target.nodeName === 'HTML') {
+          return;
+        }
+
+        isGlobalFocusVisible = false;
+
+        // Focus should stop being visible if a pointer is used on the element
+        // after it was focused using a keyboard.
+        if (
+          state.focusTarget === getEventCurrentTarget(event, context) &&
+          (type === 'mousedown' ||
+            type === 'touchstart' ||
+            type === 'pointerdown')
+        ) {
+          dispatchFocusVisibleOutEvent(context, props, state);
+        }
+        break;
+      }
+
+      case 'keydown':
+      case 'keypress':
+      case 'keyup': {
+        const nativeEvent = event.nativeEvent;
+        if (
+          nativeEvent.key === 'Tab' &&
+          !(nativeEvent.metaKey || nativeEvent.altKey || nativeEvent.ctrlKey)
+        ) {
+          isGlobalFocusVisible = true;
```

Comment:
It's probably not ideal to have global focus state being checked / modified in every instance. Should this be handled by a module outside of Flare (using native APIs) that we query for global focus visibility info? Any other ideas?

Reply:
We could add a `context.isEventComponentWithinResponderScope` API that checks if a parent event component in the tree has the same responder and bail-out of this logic?